pkg/provider/azure_storageaccount.go

```diff
@@ -109,6 +112,17 @@ func (az *Cloud) getStorageAccounts(ctx context.Context, accountOptions *Account
 				isPrivateEndpointAsExpected(acct, accountOptions)) {
 				continue
 			}
+
+			property, err := az.BlobClient.GetServiceProperties(ctx, accountOptions.SubscriptionID, accountOptions.ResourceGroup, *acct.Name)
```

Comment:
this is calling `GetServiceProperties` for every account search, that's expensive, it's better if only BlobVersioning is enabled, then call `GetServiceProperties`

Reply:
here we need to search an existed storageaccount which has the same property value of EnableBlobVersion(enable/disable) as specified by storageclass.

Reply:
since `ServiceProperties` is only used in these 3 functions, I think it would be better to only call if user enables `SoftDelete` functions, otherwise that's two expensive call in every account matching.